lib/ansible/module_utils/facts/network/fc_wwn.py

```diff
@@ -43,4 +43,19 @@ def collect(self, module=None, collected_facts=None):
             for fcfile in glob.glob('/sys/class/fc_host/*/port_name'):
                 for line in get_file_lines(fcfile):
                     fc_facts['fibre_channel_wwn'].append(line.rstrip()[2:])
+        elif sys.platform.startswith('sunos'):
+            """
+            on solaris 10 or solaris 11 should use `fcinfo hba-port`
+            on solaris 9, `prtconf -pv`
+            """
+            cmd = "/usr/sbin/fcinfo hba-port | grep 'Port WWN'"
```

Comment:
instead of hardcoding the command, use  module.get_bin_path

Reply:
changed a bit , is it ok now ?! thanks